core/commands/SubprojectDeleteCommand.php

```diff
@@ -0,0 +1,116 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+require_api( 'constant_inc.php' );
+require_api( 'config_api.php' );
+require_api( 'helper_api.php' );
+require_api( 'project_api.php' );
+require_api( 'project_hierarchy_api.php' );
+
+use Mantis\Exceptions\ClientException;
+
+/**
+ * A command that deletes a project subproject.
+ */
+class SubprojectDeleteCommand extends Command {
+	/**
+	 * $p_data['query'] is expected to contain:
+	 * - project_id (integer)
+	 * - subproject_id (integer)
+	 *
+	 * @param array $p_data The command data.
+	 */
+
+	/**
+	 * @var integer
+	 */
+	private $project_id;
+
+	/**
+	 * @var integer
+	 */
+	private $subproject_id;
+
+	/**
+	 * Constructor
+	 *
+	 * @param array $p_data The command data.
+	 */
+	function __construct( array $p_data ) {
+		parent::__construct( $p_data );
+	}
+
+	/**
+	 * Validate the data.
+	 */
+	function validate() {		
+		$this->project_id = helper_parse_id( $this->query( 'project_id' ), 'project_id' );
+
+		if( !access_has_project_level( config_get( 'manage_project_threshold' ), $this->project_id ) ) {
```

Comment:
Similar to sub-project add, we should check if we should check that subproject is manageable by the user as well.